torchmetrics/classification/accuracy.py

```diff
@@ -240,6 +241,8 @@ def update(self, preds: Tensor, target: Tensor) -> None:
             self.correct += correct
             self.total += total
         else:
+            if not self.mode:
+                raise RuntimeError("You have to have determined mode.")
```

Comment:
Is this even possible based on the conditions before?

Reply:
yes if you call compute without any update

Reply:
But we prevent that in the `__init__` of the super class I think where we assert that update was called, no?

Reply:
yes, by mypy dos not recognise this logic...